src/app/shared/store/registration.state.ts

```diff
@@ -152,21 +152,25 @@ export class RegistrationState {
   }
 
   @Action(CheckAuth)
-  checkAuth({ patchState, dispatch }: StateContext<RegistrationStateModel>): void {
-    this.oidcSecurityService.checkAuth().subscribe((auth: LoginResponse) => {
-      patchState({ isAuthorized: auth.isAuthenticated });
-      if (auth.isAuthenticated) {
-        this.oidcSecurityService.getAccessToken().subscribe((value: string) => {
-          const token = jwt_decode(value) as TokenPayload;
-          const subrole = token.subrole;
-          const role = token.role;
-          patchState({ subrole, role });
-          dispatch(new GetUserPersonalInfo()).subscribe(() => dispatch(new CheckRegistration()));
-        });
-      } else {
-        patchState({ role: Role.unauthorized, isAuthorizationLoading: false });
-      }
-    });
+  checkAuth({ patchState, dispatch }: StateContext<RegistrationStateModel>): Observable<any> {
+    return this.oidcSecurityService.checkAuth().pipe(
+      switchMap((auth: LoginResponse) => {
+        patchState({ isAuthorized: auth.isAuthenticated });
+        if (auth.isAuthenticated) {
+          return this.oidcSecurityService.getAccessToken().pipe(
+            switchMap((value: string) => {
+              const token = jwt_decode(value) as TokenPayload;
+              const subrole = token.subrole;
+              const role = token.role;
+              patchState({ subrole, role });
+              return dispatch(new GetUserPersonalInfo()).pipe(switchMap(() => dispatch(new CheckRegistration())));
+            })
+          );
+        } else {
+          patchState({ role: Role.unauthorized, isAuthorizationLoading: false });
```

Comment:
Make sure there are no errors if the user is not authenticated

Reply:
Checked again,It looks fine